Makefile

```diff
@@ -405,14 +418,22 @@ endef
 # ---------------------------------------------------------------------------------
 # This invocation will generate makefile-snippets specifying the dependency of
 # each unit-test binary on its sources and required objects. This way, we do not
-# have to enumerate dependency for each unit-test binary built, such as:
+# have to enumerate dependency for each unit-test binary built, how it's been
+# hand-defined for Mac/OSX. (For some reason the construction of calling the fn
+# unit_test_self_dependency() seems to not work on Mac/OSX as it does on Linux.)
 #
-# $(BINDIR)/$(UNIT_DIR)/l3_dump.py-test: $(OBJDIR)/$(UNITTESTS_DIR)/l3_dump.py-test.o \
-#                                        $(OBJDIR)/$(SRCDIR)/l3.o
 # See https://www.gnu.org/software/make/manual/html_node/Eval-Function.html
 # ---------------------------------------------------------------------------------
+ifeq ($(UNAME_S),Linux)
+
 $(foreach unit,$(UNIT_TESTBINS),$(eval $(call unit_test_self_dependency,$(unit))))
 
+else
+
+$(BINDIR)/$(UNIT_DIR)/l3_dump.py-test: $(OBJDIR)/$(UNITTESTS_DIR)/l3_dump.py-test.o \
+                                        $(OBJDIR)/$(SRCDIR)/l3.o
```

Comment:
See the comments on L422. I couldn't figure out why this inline Makefile-function is not working on Mac.

It's a minor wart; we can live with for now.